tests/nlu/featurizers/test_count_vectors_featurizer.py

```diff
@@ -52,6 +52,50 @@ def test_count_vector_featurizer(sentence, expected, expected_cls):
     assert np.all(actual[-1] == expected_cls)
 
 
+@pytest.mark.parametrize(
+    "sentence, intent, response, intent_features, response_features",
+    [("hello", "greet", None, [[1]], None), ("hello", "greet", "hi", [[1]], [[1]])],
+)
+def test_count_vector_featurizer_response_attribute_featurization(
+    sentence, intent, response, intent_features, response_features
+):
+    ftr = CountVectorsFeaturizer({"token_pattern": r"(?u)\b\w+\b"})
+    tk = WhitespaceTokenizer()
+
+    train_message = Message(sentence)
+    # this is needed for a valid training example
+    train_message.set(INTENT_ATTRIBUTE, intent)
+    train_message.set(RESPONSE_ATTRIBUTE, response)
+
+    second_message = Message("hello")
+    second_message.set(RESPONSE_ATTRIBUTE, "hi")
+    second_message.set(INTENT_ATTRIBUTE, "greet")
+
+    data = TrainingData([train_message, second_message])
+
+    tk.train(data)
+    ftr.train(data)
+
+    if intent_features:
+        assert (
+            train_message.get(SPARSE_FEATURE_NAMES[INTENT_ATTRIBUTE]).toarray()[0]
+            == intent_features
+        )
+    else:
+        assert train_message.get(SPARSE_FEATURE_NAMES[INTENT_ATTRIBUTE]) is None
+
+    if response_features:
+        assert (
+            train_message.get(SPARSE_FEATURE_NAMES[RESPONSE_ATTRIBUTE]).toarray()[0]
+            == response_features
+        )
+    else:
+        assert train_message.get(SPARSE_FEATURE_NAMES[RESPONSE_ATTRIBUTE]).shape == (
```

Comment:
Shouldn't we also be checking that sparse features for `RESPONSE_ATTRIBUTE` is actually None in case `response_features` was None?
If `RESPONSE` attribute is None then all corresponding attributes for `RESPONSE` should also be None.

Reply:
We can also do it like this, I guess. Currently, all other featurizers will just return an empty array. Will check if `None` also works.

Reply:
Yes, programmatically it should work. Then it's a minor suggestion from my side.

Reply:
Shouldn't the shape of empty array be (0,)?